src/components/autocompleter/autocompleter.test.jsx

```diff
@@ -0,0 +1,139 @@
+import React, { createRef } from 'react';
+import {
+  render, screen,
+} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import jestServer from '../../mocks/jest-server.js';
+import Autocompleter from './autocompleter.jsx';
+import mockHandlers from './mock-handlers.js';
+
+describe('src/components/autocompleter/autocompleter', () => {
+  beforeEach(() => {
+    jestServer.use(...mockHandlers());
+  });
+
+  const requiredProps = {
+    id: 'test-id',
+    label: 'Test label',
+    name: 'field-id',
+    apiEndpoint: '/models',
+  };
+
+  it('has defaults', () => {
+    const ref = createRef();
+    render(<Autocompleter {...requiredProps} ref={ref} />);
+    expect(document.body).toMatchSnapshot();
+    expect(ref.current).toMatchSnapshot();
+  });
+
+  describe('className API', () => {
+    it('prioritizes className prop', () => {
+      const { container } = render(<Autocompleter {...requiredProps} className="prioritize-me" />);
+      expect(container.firstChild).toHaveClass('prioritize-me');
+    });
+  });
+
+  describe('errorText', () => {
+    it('sets the input to be invalid', () => {
+      render(<Autocompleter {...requiredProps} errorText="not valid" />);
+      expect(screen.getByLabelText('Test label')).not.toBeValid();
+    });
+  });
+
+  describe('id API', () => {
+    it('accepts an ID', () => {
+      render(<Autocompleter {...requiredProps} id="this-is-an-id" />);
+      expect(screen.getByLabelText('Test label')).toHaveAttribute('id', 'this-is-an-id');
+    });
+  });
+
+  describe('label API', () => {
+    it('accepts a label', () => {
+      render(<Autocompleter {...requiredProps} label="Bar" />);
+      expect(screen.getByLabelText('Bar')).toBeDefined();
+    });
+  });
+
+  describe('placeholder API', () => {
+    it('accepts a placeholder', () => {
+      render(<Autocompleter {...requiredProps} placeholder="I am place" />);
+      expect(screen.getByLabelText('Test label')).toHaveAttribute('placeholder', 'I am place');
+    });
+  });
+
+  describe('readOnly API', () => {
+    it('sets the readonly attribute', () => {
+      render(<Autocompleter {...requiredProps} readOnly />);
+      expect(screen.getByLabelText('Test label')).toHaveAttribute('readonly', '');
+    });
+
+    it('unsets the readonly attribute', () => {
+      render(<Autocompleter {...requiredProps} readOnly={false} />);
+      expect(screen.getByLabelText('Test label')).not.toHaveAttribute('readonly', '');
+    });
+  });
+
+  describe('required API', () => {
+    it('sets the required attribute', () => {
+      render(<Autocompleter {...requiredProps} required />);
+      expect(screen.getByLabelText('Test label')).toBeRequired();
+    });
+
+    it('unsets the required attribute', () => {
+      render(<Autocompleter {...requiredProps} required={false} />);
+      expect(screen.getByLabelText('Test label')).not.toBeRequired();
+    });
+  });
+
+  describe('value API', () => {
```

Comment:
We want to add a test about re-rendering:

```js
it('accepts a value', async () => {
      const value1 = { id: 22, name: 'cool dude' };
      const value2 = { id: 68, name: 'coolio dude' };
      const { rerender } = render(<Autocompleter {...requiredProps} value={value1} />);
      expect(screen.getByLabelText('Test label')).toHaveValue('cool dude');
      rerender(<Autocompleter {...requiredProps} value={value2} />);
      expect(screen.getByLabelText('Test label')).toHaveValue('coolio dude');
    });
```

Reply:
Is this testing React-core, is there any reason to think this wouldn't work?